apps/music/js/ui/views/list_view.js

```diff
@@ -75,9 +75,8 @@ var ListView = {
 
   hideSearch: function lv_hideSearch() {
     this.searchInput.value = '';
-    // XXX: we probably want to animate this...
-    if (this.view.scrollTop < this.searchBox.offsetHeight) {
-      this.view.scrollTop = this.searchBox.offsetHeight;
+    if (this.view.scrollTop < this.anchor.offsetTop) {
+      this.anchor.scrollIntoView({behavior: 'smooth'});
```

Comment:
@dominickuo Sadly, it's breaking some of our tests, since b2g-desktop's implementation of `scrollIntoView` actually scrolls the *app* off-screen. I'm trying to figure out how to fix that.